tests/testsuite/build.rs

```diff
@@ -1494,6 +1497,10 @@ fn crate_env_vars() {
                     // Check that CARGO_TARGET_TMPDIR isn't set for unit tests
                     assert!(option_env!("CARGO_TARGET_TMPDIR").is_none());
                     env::var("CARGO_TARGET_TMPDIR").unwrap_err();
+
+                    // Check that CARGO_WORKSPACE_DIR isn't set for unit tests
+                    assert!(option_env!("CARGO_WORKSPACE_DIR").is_none());
+                    env::var("CARGO_WORKSPACE_DIR").unwrap_err();
```

Comment:
Should we be making unit tests work?

Reply:
Do you also want to update the behavior of `CARGO_BIN_EXE_{name}` and `CARGO_TARGET_TMPDIR`? They currently also only work with integration tests and benchmarks. Otherwise I think the behavior of those env vars would be inconsistent.

Reply:
If we decide to do it, perhaps we should do it in a separate PR?

Reply:
I'd recommend leaving this as an open question (like the name) for us to hash out during stabilization.  If you haven't updated for unit tests, maybe we leave it as-is for now?

Reply:
I said that but...

It looks like `CARGO_BIN_EXE_` and [`CARGO_TARGET_TMPDIR`](https://github.com/rust-lang/cargo/pull/9375) are for integration tests because, in theory, unit tests wouldn't be needing to do operations like this. However, it perfectly makes sense to have basic snapshots in unit tests, so I would lean towards supporting this in unit tests.  I'd also be in favor of re-evaluating `CARGO_TARGET_TMPDIR` so it can be used in unit tests but thats another topic.